plugins/modules/storagebox_subaccount.py

```diff
@@ -0,0 +1,473 @@
+        # function is only called when storagebox and subaccount were already verified
+        if error is not None:  # pragma: no cover
+            raise AssertionError('Unhandled error happened during password update of {0}'.format(self.username))  # pragma: no cover
+
+    def mgmt_url(self):
+        return "{0}/storagebox/{1}/subaccount/{2}".format(
+            BASE_URL, self.storagebox_id, self.username
+        )
+
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):  # pragma: no cover
+            return False  # pragma: no cover
+        return self.__dict__ == other.__dict__
+
+    # Python 2, I hate you
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+
+def get_subaccounts(module, storagebox_id):
+    url = "{0}/storagebox/{1}/subaccount".format(BASE_URL, storagebox_id)
+    result, error = fetch_url_json(module, url, accept_errors=["STORAGEBOX_NOT_FOUND"])
+    if error:
+        module.fail_json(
+            msg="Storagebox with ID {0} does not exist".format(storagebox_id)
+        )
+    return [Subaccount(storagebox_id, **item["subaccount"]) for item in result]
+
+
+def main():
+    argument_spec = dict(
+        storagebox_id=dict(type="int", required=True),
+        subaccount=dict(
+            type="dict",
+            required=True,
+            options=dict(
+                state=dict(type="str", default="present", choices=["present", "absent"]),
+                username=dict(type="str"),
+                password=dict(type="str", no_log=True),
+                homedirectory=dict(type="str"),
+                samba=dict(type="bool", default=False),
+                ssh=dict(type="bool", default=False),
+                external_reachability=dict(type="bool", default=False),
+                webdav=dict(type="bool", default=False),
+                readonly=dict(type="bool", default=False),
+                comment=dict(type="str", default=""),
+                idempotence=dict(type="str", choices=["username", "comment"], default="username"),
+            ),
+        ),
+        password_mode=dict(
+            type="str",
+            no_log=True,
+            choices=["update-if-provided", "ignore-if-exists", "set-to-random"],
+            default="update-if-provided",
+        ),
+    )
+    argument_spec.update(ROBOT_DEFAULT_ARGUMENT_SPEC)
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
+
+    check_mode = module.check_mode
+    storagebox_id = module.params["storagebox_id"]
+    subaccount = module.params["subaccount"]
+    password_mode = module.params['password_mode']
+
+    account_identifier = subaccount[subaccount["idempotence"]]
+
+    existing_subaccounts = get_subaccounts(module, storagebox_id)
+    existing = next(
+        (
+            sa for sa in existing_subaccounts
+            if getattr(sa, subaccount["idempotence"], None) == account_identifier
+        ),
+        None,
+    )
```

Comment:
You need to check that there is at most one subaccount matching the identifier (at least if `module.params["idempotence"] == "comment"`).